exchanges/lbank/lbank.go

```diff
@@ -0,0 +1,514 @@
+	exchange.Base
+	privateKey    *rsa.PrivateKey
+	privKeyLoaded bool
+	WebsocketConn *websocket.Conn
+}
+
+const (
+	lbankAPIURL          = "https://api.lbkex.com"
+	lbankAPIVersion      = "1"
+	lbankAuthRateLimit   = 0
+	lbankUnAuthRateLimit = 0
+
+	// Public endpoints
+	lbankTicker         = "ticker.do"
+	lbankCurrencyPairs  = "currencyPairs.do"
+	lbankMarketDepths   = "depth.do"
+	lbankTrades         = "trades.do"
+	lbankKlines         = "kline.do"
+	lbankPairInfo       = "accuracy.do"
+	lbankUSD2CNYRate    = "usdToCny.do"
+	lbankWithdrawConfig = "withdrawConfigs.do"
+
+	// Authenticated endpoints
+	lbankUserInfo                = "user_info.do"
+	lbankPlaceOrder              = "create_order.do"
+	lbankCancelOrder             = "cancel_order.do"
+	lbankQueryOrder              = "orders_info.do"
+	lbankQueryHistoryOrder       = "orders_info_history.do"
+	lbankOrderTransactionDetails = "order_transaction_detail.do"
+	lbankPastTransactions        = "transaction_history.do"
+	lbankOpeningOrders           = "orders_info_no_deal.do"
+	lbankWithdrawalRecords       = "withdraws.do"
+	lbankWithdraw                = "withdraw.do"
+	lbankRevokeWithdraw          = "withdrawCancel.do"
+)
+
+// SetDefaults sets the basic defaults for Lbank
+func (l *Lbank) SetDefaults() {
+	l.Name = "Lbank"
+	l.RESTPollingDelay = 10
+	l.RequestCurrencyPairFormat.Delimiter = "_"
+	l.ConfigCurrencyPairFormat.Delimiter = "_"
+	l.AssetTypes = []string{ticker.Spot}
+	l.Requester = request.New(l.Name,
+		request.NewRateLimit(time.Second, lbankAuthRateLimit),
+		request.NewRateLimit(time.Second, lbankUnAuthRateLimit),
+		common.NewHTTPClientWithTimeout(exchange.DefaultHTTPTimeout))
+	l.APIUrlDefault = lbankAPIURL
+	l.APIUrl = l.APIUrlDefault
+	l.WebsocketInit()
+}
+
+// Setup takes in the supplied exchange configuration details and sets params
+func (l *Lbank) Setup(exch *config.ExchangeConfig) {
+	if !exch.Enabled {
+		l.SetEnabled(false)
+	} else {
+		l.Enabled = true
+		l.AuthenticatedAPISupport = exch.AuthenticatedAPISupport
+		l.AuthenticatedWebsocketAPISupport = exch.AuthenticatedWebsocketAPISupport
+		l.SetAPIKeys(exch.APIKey, exch.APISecret, "", false)
+		l.SetHTTPClientTimeout(exch.HTTPTimeout)
+		l.SetHTTPClientUserAgent(exch.HTTPUserAgent)
+		l.RESTPollingDelay = exch.RESTPollingDelay
+		l.Verbose = exch.Verbose
+		l.Websocket.SetWsStatusAndConnection(exch.Websocket)
+		l.BaseCurrencies = exch.BaseCurrencies
+		l.AvailablePairs = exch.AvailablePairs
+		l.EnabledPairs = exch.EnabledPairs
+		err := l.SetCurrencyPairFormat()
+		if err != nil {
+			log.Fatal(err)
+		}
+		err = l.SetAssetTypes()
+		if err != nil {
+			log.Fatal(err)
+		}
+		err = l.SetAutoPairDefaults()
+		if err != nil {
+			log.Fatal(err)
+		}
+		err = l.SetAPIURL(exch)
+		if err != nil {
+			log.Fatal(err)
+		}
+		err = l.SetClientProxyAddress(exch.ProxyAddress)
+		if err != nil {
+			log.Fatal(err)
+		}
+		if l.AuthenticatedAPISupport {
+			err = l.loadPrivKey()
+			if err != nil {
+				log.Fatal(err)
```

Comment:
Instead of Fataling here if the private key fails to load we can set AuthenticatedAPISupport back to false and log.Errorf() letting the user know it failed and continue on as normal